tests/test_connection.py

```diff
@@ -4,17 +4,15 @@
 
 import pytest
 
-from aioredis.connection import UnixDomainSocketConnection
+from aioredis.connection import PythonParser, UnixDomainSocketConnection
 from aioredis.exceptions import InvalidResponse
-from aioredis.utils import HIREDIS_AVAILABLE
 
-if TYPE_CHECKING:
-    from aioredis.connection import PythonParser
 
-
-@pytest.mark.skipif(HIREDIS_AVAILABLE, reason="PythonParser only")
 @pytest.mark.asyncio
-async def test_invalid_response(r):
+@pytest.mark.parametrize("create_redis", [(True, PythonParser)], indirect=True)
+async def test_invalid_response(create_redis):
```

Comment:
This test seems like it wasn't run for some time, as in the case of connection pool it will fail with `AttributeError`, because only a single connection client has a connection set

Reply:
And it also didn't work on py =< 3.7

Reply:
> And it also didn't work on py =< 3.7

We should probably drop Python 3.6 now anyways since it's EOL soon.

Reply:
But for 3.7 it's still a problem and pypy seems to be unhappy about this test also.
AsyncMock solves this problem.